wallet-new/server/Cardano/Wallet/API/V1/Handlers/Accounts.hs

```diff
@@ -57,23 +57,17 @@ listAccounts RequestParams {..} = do
       }
     _ -> return $ OneOf $ Left example
 
--- | This is an example of how POST requests might look like.
--- It also shows an example of how an error might look like.
--- NOTE: This will probably change drastically as soon as we start using our
--- custom monad as a base of the Handler stack, so the example here is just to
--- give the idea of how it will look like on Swagger.
-newAccount :: WalletId -> Maybe Text -> AccountUpdate -> MonadV1 Account
-newAccount wId _ AccountUpdate{..} = do
-    -- In real code we would generate things like addresses (if needed) or
-    -- any other form of Id/data.
-    newId <- liftIO $ generate arbitrary
-    return Account
-        { accId = newId
-        , accAmount = Core.mkCoin 0
-        , accAddresses = mempty
-        , accName = uaccName
-        , accWalletId = wId
-        }
+newAccount
+    :: (MonadThrow m, V0.MonadWalletLogic ctx m)
+    => WalletId -> Maybe PassPhrase -> AccountUpdate -> m Account
+newAccount wId mPassPhrase accUpdate =
+    case mPassPhrase of
+        Nothing -> throwM $ Errors.toError Errors.NoPassPhrase
```

Comment:
Are we sure this is correct? AFAIK a `PassPhrase` is completely optional and we shouldn't be failing if the wallet was created initially with no `PassPhrase`.

Also, I know this might sound confusing, but we have renamed `PassPhrase` to `SpendingPassword` as it's one of the most confusing points of the old API. There is a type synonym in `Cardano.Wallet.API.V1.Types`. Would you mind using it for external-facing code? Thanks a ton! 😉

Reply:
@adinapoli-iohk `V0.newAccount` (https://github.com/input-output-hk/cardano-sl/blob/master/wallet/src/Pos/Wallet/Web/Methods/Logic.hs#L210-L212) does need a `PassPhrase`, so it seems that `PassPhrase` is not optional. Could you give me more details, pls?

Reply:
Ah, I see... We have similar "issue" solved in `V1.newWallet`. Pls ignore my previous question ^, @adinapoli-iohk